src/awkward/_reducers.py

```diff
@@ -34,8 +34,17 @@ def preferred_dtype(self) -> DTypeLike:
     def highlevel_function(cls):
         return getattr(ak.operations, cls.name)
 
+    @abstractmethod
+    def apply(
+        self,
+        array: ak.contents.NumpyArray,
+        parents: ak.index.Index,
+        outlength: ShapeItem,
+    ) -> ak.contents.NumpyArray:
+        raise NotImplementedError
+
     @classmethod
-    def return_dtype(cls, given_dtype: DTypeLike) -> DTypeLike:
+    def _return_dtype(cls, given_dtype: DTypeLike) -> DTypeLike:
```

Comment:
These underscored methods are internal-only, so they should be private.

Reply:
That's true. If we were exposing the `Reducer` classes at all, the `return_dtype` is something one might want to know about it, but they're in a hidden module.

Reply:
Actually, it's because they're in a hidden module that I didn't bother to hide the methods. There only needs to be one underscore between `ak.` and object. Maybe the distinction you're making here is between L3 and L4?